jetbrains-core/tst/software/aws/toolkits/jetbrains/ui/ResourceSelectorTest.kt

```diff
@@ -59,6 +60,32 @@ class ResourceSelectorTest {
         assertThat(comboBox.isEnabled).isEqualTo(false)
     }
 
+    @Test
+    fun comboBoxPopulation_updateStateToDesired() {
```

Comment:
This test confuses me.

What puts the loading status into SUCCESSFUL? There is a while loop blocking the unlock call while the populateValues runs in a loop while the lock is locked. So how is it unlocked, so that it loads?

Reply:
The initial status for `ResourceSelector` is `SUCCESSFUL` and the status could be changed into `SUCCESSFUL` after the resources are loaded successfully (ie. the `block` call has been ran successfully).

When loading resources, the status changes as `SUCCESSFUL(the initial status) -> LOADING -> SUCCESSFUL`. We need to wait the status to change to `LOADING` until we call `enable`. That is why a `while` loop is there waiting for the status to change to `LOADING`. When calling `enable`, we need to make sure the status is in `LOADING` and the lock is meant to act as loading the resources so that after unlocking, the resources are loaded.

Reply:
It seems to conflate 2 ideas into 1 boolean.

From an API perspective, the combo box has 2 states. isEnabled and isLoading. 
The actual UI element should be disabled if isEnabled == false || isLoading == true.

The fact that this code exists:
```
 comboBox.isEnabled = true
 assertThat(comboBox.isEnabled).isEqualTo(false)
``` 
is super confusing. I just set it to enabled and yet it is still false.

Reply:
Cut #726 to revisit the design of this combo box